client/extensions/woocommerce/woocommerce-services/lib/utils/print-document.js

```diff
@@ -55,12 +55,24 @@ const buildBlob = ( b64Content, mimeType ) => {
  * @returns {Promise} Promise that resolves if the printing dialog (or equivalent) was correctly
  * invoked, rejects otherwise.
  */
-export default ( { b64Content, mimeType }, fileName ) => {
+export default ( { b64Content, mimeType }, fileName, download = false ) => {
 	const blob = buildBlob( b64Content, mimeType );
-	const blobUrl = 'ie' !== getPDFSupport() ? URL.createObjectURL( blob ) : null; // IE has no use for "blob:" URLs
+	const blobUrl = 'ie' !== getPDFSupport( download ) ? URL.createObjectURL( blob ) : null; // IE has no use for "blob:" URLs
 
-	switch ( getPDFSupport() ) {
+	switch ( getPDFSupport( download ) ) {
 		case 'native':
+			if ( download ) {
+				const link = document.createElement( 'a' );
+				link.setAttribute( 'href', blobUrl );
+				link.setAttribute( 'download', fileName );
+
+				const event = document.createEvent( 'MouseEvents' );
+				event.initMouseEvent( 'click', true, true, window, 1, 0, 0, 0, 0, false, false, false, false, 0, null );
```

Comment:
Wow, this just looks like the Konami code 😂 
Where did you get this code from? I'd like to understand it.